### DISCUSSION TITLE
How to automatically clean up `_freeze` folder

### DISCUSSION THREAD

**Discussion by pipme** [quarto-dev/quarto-cli#5937] [Q&A] [answered]
### Description

`_freeze` folder seems to retain old results. For example, when the `qmd` file or its parent folder is renamed, it stills keeps the previously frozen contents.

Is there a way to clean up it automatically when rendering?

Thanks!

**Comment by mcanouil** [comment]
Currently, there is no native way within Quarto, but you could write a script as a pre-render script (not perfect).

@cscheid Do you have any additional information? Is it something that can be done within Quarto, like checking if there still is a file associated with the freeze directories?

**Reply by cscheid** [reply, marked as answer]
I think it's currently hard to do under file moves; quarto doesn't know how guess what file name some arbitrary new file used to have.

It might be possible in 1.5, when we plan to store metadata about project compilation (so we know what .md file is associated with an input .qmd).

**Reply by pipme** [reply]
Good to know, thanks!
